base/linalg/lu.jl

```diff
@@ -179,36 +179,13 @@ function logabsdet{T,S}(A::LU{T,S})  # return log(abs(det)) and sign(det)
         end
         abs_det += log(abs(dg_ii))
     end
-    s = (isodd(c) ? -one(real(T)) : one(real(T))) * P
+    s = (isodd(c) ? -one(T) : one(T)) * P
```

Comment:
Please check my comments again. I've pulled your branch and made the changes and I don't see the test error. It's `abs_det`, not `P`, that should have the `real`.